dialog/input_test.go

```diff
@@ -0,0 +1,21 @@
+package dialog
+
+import (
+	"testing"
+
+	"fyne.io/fyne/test"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestInputDialogConfirm(t *testing.T) {
+	ch := make(chan string)
+	i := NewInput("test1", "test2", func(s string) { ch <- s }, test.NewWindow(nil))
+	i.Show()
+
+	assert.False(t, i.win.Hidden)
+	test.Type(i.entry, "test3")
+	go test.Tap(i.confirmButton)
```

Comment:
I think the tests work with synchronos tapping - did it not work without the goroutine?

Reply:
I cargo-cult-ed some of the other tests which used goroutines, I can switch to synchronous if that is preferred.

Reply:
Ah well, if it is consistent just leave it ;)
When we review dialogs properly this could more easily get revised with everything else.